packages/mattermost-redux/src/selectors/entities/cloud.ts

```diff
@@ -35,6 +40,15 @@ export function getSubscriptionProduct(state: GlobalState): Product | undefined
     return products[subscription.product_id];
 }
 
+export function checkSubscriptionIsLegacyFree(state: GlobalState): boolean {
+    return Boolean(LegacyFreeProductIds[getCloudSubscription(state)?.product_id || '']);
+}
+
+export function checkHadPriorTrial(state: GlobalState): boolean {
```

Comment:
this is nice, will use it to replace other places where the same validation is made without the selector. Thanks!